### PR TITLE
Add an example for JPA based idempotent repository

### PR DESCRIPTION
**Pull request by aldettinger**
Note that the `main` branch points at the latest stable Camel Quarkus release.
Pull requests should be generally send against the `camel-quarkus-main` branch pointing at the current Camel Quarkus SNAPSHOT.

### CODE REVIEW

**Comment by aldettinger** [comment]
As a follow up, the doc release guide should be updated with `./mvnw process-sources -f jpa-idempotent-repository/`.

**Review by oscerd — APPROVED** [review]

File: jpa-idempotent-repository/pom.xml [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+	<artifactId>camel-quarkus-examples-jpa-idempotent-repository</artifactId>
+	<groupId>org.apache.camel.quarkus.examples</groupId>
+	<version>3.5.0-SNAPSHOT</version>
+
+	<name>Camel Quarkus :: Examples :: JPA Idempotent Repository</name>
+	<description>Camel Quarkus Example :: JPA Idempotent Repository</description>
+
+	<properties>
+		<quarkus.platform.version>3.4.1</quarkus.platform.version>
+		<camel-quarkus.platform.version>3.5.0-SNAPSHOT</camel-quarkus.platform.version>
```
Comment (aldettinger) [inline jpa-idempotent-repository/pom.xml:32]: Set this version here similarly to other example in the camel-quarkus-main branch. It's not working locally.

Comment (aldettinger) [reply jpa-idempotent-repository/pom.xml:32]: ok, got it.

File: jpa-idempotent-repository/pom.xml [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+					<execution>
+						<id>sync-k8s-resource-versions</id>
+						<inherited>false</inherited>
+						<goals>
+							<goal>execute</goal>
+						</goals>
+						<phase>process-sources</phase>
+						<configuration>
+							<source>
+                                import java.util.regex.Matcher
```
Comment (jamesnetherton) [inline jpa-idempotent-repository/pom.xml:295]: Nitpick - indentation.

File: jpa-idempotent-repository/src/main/java/org/acme/jpa/idempotent/repository/CostlyApiService.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+public class CostlyApiService {
+
+    private static final Logger LOG = Logger.getLogger(CostlyApiService.class);
+
+    private static Set<String> ALREADY_USED_CONTENT = ConcurrentHashMap.newKeySet();
+
+    /**
+     * The content parameter is populated with the incoming HTTP body sent to this API.
+     */
+    @Handler
```
Comment (jamesnetherton) [inline jpa-idempotent-repository/src/main/java/org/acme/jpa/idempotent/repository/CostlyApiService.java:40]: Thanks for pointing me to `@Handler` :slightly_smiling_face:.

I opened https://github.com/apache/camel-quarkus/issues/5378 as a future improvement.

**Review by jamesnetherton — APPROVED** [review]
LGTM - I left a nitpick comment.

File: jpa-idempotent-repository/src/main/java/org/acme/jpa/idempotent/repository/CostlyApiService.java (hunk 3 above)
Comment (aldettinger) [reply jpa-idempotent-repository/src/main/java/org/acme/jpa/idempotent/repository/CostlyApiService.java:40]: Yes, good idea.

File: jpa-idempotent-repository/pom.xml (hunk 2 above)
Comment (aldettinger) [reply jpa-idempotent-repository/pom.xml:290]: Nice catch, probably an undue pom editor replace messing up the whole indentation. It's fixed now.

**Pull request merged**